src/program.cpp

```diff
@@ -1682,12 +1680,24 @@ cl_int cvk_program::build(build_operation operation, cl_uint num_devices,
     m_operation_callback_data = data;
 
     cl_int ret = CL_SUCCESS;
-    if (cb) {
+    bool build_in_separate_thread = config.build_in_separate_thread() || cb;
+    bool check_build_status = !cb;
```

Comment:
```suggestion
    bool wait_for_completion = !cb;
```
Feels a bit clearer maybe.